documentation/assemblies/upgrading/assembly-upgrade-resources.adoc

```diff
@@ -3,52 +3,41 @@
+[role="_abstract"]
+After you have upgraded Strimzi to {ProductVersion}, you must ensure that your custom resources are using API version `{ApiVersion}`.
+You can do this any time after upgrading to {ProductVersion}, but the upgrades must be completed before the next Strimzi minor version update.
 
-* `Kafka`
-* `KafkaConnect`
-* `KafkaConnectS2I`
-* `KafkaConnector`
-* `KafkaMirrorMaker`
-* `KafkaMirrorMaker2`
-* `KafkaTopic`
-* `KafkaUser`
-* `KafkaBridge`
-* `KafkaRebalance`
+IMPORTANT: Upgrade of the custom resources to `{ApiVersion}` _must_ be performed after xref:assembly-upgrade-{context}[upgrading the Cluster Operator], so the Cluster Operator can understand the resources.
 
-The `v1beta1` API version is deprecated for the following Strimzi resources:
+NOTE: Upgrade of the custom resources to `{ApiVersion}` prepares Strimzi for a move to Kubernetes CRD `v1`,
+which is required for Kubernetes 1.22.
 
-* `Kafka`
-* `KafkaConnect`
-* `KafkaConnectS2I`
-* `KafkaMirrorMaker`
-* `KafkaTopic`
-* `KafkaUser`
+.CLI upgrades to custom resources
 
-Update these resources to use the `{KafkaApiVersion}` API version.
+Strimzi provides an _API conversion tool_ with its release artifacts.
 
-This section describes the upgrade steps for the resources.
+ifdef::Downloading[]
+You can download its ZIP or TAR.GZ file from {ReleaseDownload}.
+endif::Downloading[]
+ifndef::Downloading[]
+You can download its ZIP or TAR.GZ from {ZipDownload}.
+endif::Downloading[]
+To use the tool, extract it and use the scripts in the `bin` directory.
 
-IMPORTANT: The upgrade of resources _must_ be performed after xref:assembly-upgrade-{context}[upgrading the Cluster Operator], so the Cluster Operator can understand the resources.
+From its CLI, you can then use the tool to convert the format of your custom resources to `{ApiVersion}` in one of two ways:
 
-.What if the resource upgrade does not take effect?
-If the upgrade does not take effect, a warning is given in the logs on reconciliation to indicate that the resource cannot be updated until the `apiVersion` is updated.
+* xref:proc-upgrade-cli-tool-files-{context}[]
+* xref:proc-upgrade-cli-tool-direct-{context}[]
 
-To trigger the update, make a cosmetic change to the custom resource, such as adding an annotation.
+After the conversion of your custom resources, you can also set `{ApiVersion}` as the _storage_ API version in your CRDs:
 
-Example annotation:
+* xref:proc-upgrade-cli-tool-crds-{context}[]
 
-[source,shell,subs="+quotes,attributes"]
-----
-metadata:
-  # ...
-  annotations:
-    upgrade: "Upgraded to {KafkaApiVersion}"
-----
+.Manual upgrades to custom resources
 
-The following procedures describe the steps to update specific resources to use the `{KafkaApiVersion}` API version:
+Instead of using the API conversion tool to update custom resources to `{ApiVersion}`, you can manually update each custom resource to use `{ApiVersion}`:
 
 * xref:proc-upgrade-kafka-resources-{context}[]
```

Comment:
I think we need to update this list as the last PR from @laidan6000 added more of these.

Reply:
I've updated as per merged pr